lib/index.js

```diff
@@ -10,20 +13,38 @@ function createPreprocessor(args, config, logger, helper) {
   var log = logger.create('preprocessor.babel');
   config = config || {};
 
-  function preprocess(content, file, done) {
-    log.debug('Processing "%s".', file.originalPath);
+  var cachePath = config.cachePath;
+  var cache = null;
+  if (cachePath) {
+    cache = new LocalCache(cachePath);
+  }
 
+  function preprocess(content, file, done) {
     try {
       var options = createOptions(args, config, helper, file);
       file.path = options.filename || file.path;
 
-      var babelOptions = helper.merge({}, options, { filename: file.originalPath });
-      var processed = babel.transform(content, babelOptions);
-      var code = content;
-      if (processed) {
-        code = processed.code;
+      var cacheEntry = cache ? cache.getItem(file.originalPath) : null;
+      var md5 = crypto.createHash('md5').update(content).digest('hex');
+
+      if (!cacheEntry || cacheEntry.md5 !== md5) {
+        log.debug('Processing "%s" - Cache miss.', file.originalPath);
+
+        var babelOptions = helper.merge({}, options, { filename: file.originalPath });
+        var processed = babel.transform(content, babelOptions);
+        var code = content;
+        if (processed) {
+          code = processed.code;
+        }
+
+        if (cache) {
+          cache.setItem(file.originalPath, { md5: md5, code: code });
+        }
+        done(null, code);
+      } else {
+        log.debug('Processing "%s" - Cache hit.', file.originalPath);
+        done(null, cacheEntry.code);
```

Comment:
This function got a bit convoluted now. Also, the cache miss/hit messages are logged even when cache is disabled. Could you extract the caching logic to a function? For example:

```js
var transformFunc = cache ? transformWithCache : transform;
var code = transformFunc(content, file);

function transform(content, file) {
  var babelOptions = helper.merge({}, options, { filename: file.originalPath });
  var processed = babel.transform(content, babelOptions);
  return processed ? processed.code : content;
}

function transformWithCache(content, file) {
  // Uses caching and the transform function.
}
```

Reply:
Good call.